sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -425,9 +425,12 @@ object IntervalUtils {
   }
 
   private object ParseState extends Enumeration {
+    type ParseState = Value
+
     val PREFIX,
         BEGIN_VALUE,
         PARSE_SIGN,
+        TRIM_VALUE,
         PARSE_UNIT_VALUE,
         FRACTIONAL_PART,
         BEGIN_UNIT_NAME,
```

Comment:
In `BEGIN_UNIT_NAME` we also trim the spaces. Can we have a consistent way to do it? Now there are 2 ways:
1. have an intermedia state to trim the spaces
2. trim the spaces at the beginning of a state.

Reply:
we can add a `TRIM_UNIT` state to unify these 2 ways